channeld/full_channel.c

```diff
@@ -1316,16 +1351,21 @@ bool channel_force_htlcs(struct channel *channel,
 			htlc->failed_scid = NULL;
 	}
 
-	/* Now adjust balances.  The balance never goes negative, because
-	 * we do them in id order. */
+	/* Add giant offset so we never go negative here. */
+	if (!offset_balances(channel))
```

Comment:
i assume we check an HTLC's validity (in terms of amount) before this point?

Reply:
This code is only used when we're restoring HTLCs lightningd told us about.  So presumably they've been fully checked.